### ISSUE TITLE
ErrorMaskHelper: assertion error

### ISSUE THREAD

**Issue by ptrebert** [maickrau/MBG#4]
Hi Mikko,

can you diagnose this error:

```
MBG bioconda 1.0.6
Parameters: k=501,w=100,a=1,u=2,t=12,errormasking=unknown,endkmers=yes,blunt=no
Collecting end k-mers
Reading sequences from [omitted]
MBG: src/ErrorMaskHelper.cpp:87: CharType complement(CharType): Assertion `code < multiRLEReverseComplements.size()' failed.
```

For the record, I set `--error-masking collapse-msat`...

Best,
Peter

**Comment by maickrau** [comment]
Looks like a bug in MBG. Could you upload the dataset as well?

**Comment by ptrebert** [comment]
I have added the readset to the usual Globus share (~23 GB)

**Comment by ptrebert** [comment]
I could downsample the reads to < 1 GB, and this subset is still triggering the error (but even smaller subsets did not):

```
/usr/bin/time -v MBG -i NA18989_hgsvc_pbsq2-ccs_1000.ec-reads.sample.fasta -t 4 -k 501 -w 100 --error-masking collapse-msat --include-end-kmers --out test.gfa --output-sequence-paths test.paths.gaf
MBG bioconda 1.0.6
Parameters: k=501,w=100,a=1,u=2,t=4,errormasking=unknown,endkmers=yes,blunt=no
Collecting end k-mers
Reading sequences from NA18989_hgsvc_pbsq2-ccs_1000.ec-reads.sample.fasta
MBG: src/ErrorMaskHelper.cpp:87: CharType complement(CharType): Assertion `code < multiRLEReverseComplements.size()' failed.
Command terminated by signal 6
        Command being timed: "MBG -i NA18989_hgsvc_pbsq2-ccs_1000.ec-reads.sample.fasta -t 4 -k 501 -w 100 --error-masking collapse-msat --include-end-kmers --out test.gfa --output-sequence-paths test.paths.gaf"
        User time (seconds): 38.09
        System time (seconds): 0.98
        Percent of CPU this job got: 102%
        Elapsed (wall clock) time (h:mm:ss or m:ss): 0:38.28
        Average shared text size (kbytes): 0
        Average unshared data size (kbytes): 0
        Average stack size (kbytes): 0
        Average total size (kbytes): 0
        Maximum resident set size (kbytes): 533636
```

I hope this is now small enough for debugging on your laptop...

**Comment by maickrau** [comment]
This was caused by `--include-end-kmers` when the input file has sequences shorter than `k-w`, this is fixed in 47e1ff4

**Comment by ptrebert** [comment]
great, thanks
